### PR TITLE
feat(kuma-cp) new multicluster flow

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

This PR introduces a new multicluster flow.

#### Before
When user want to connect Global with Remote they had to
* Deploy Global
* Extract public IP of global and use it in Deploy Remote command
* Wait for Remote to be deployed
* Extract Remote Ingress address
* Apply zone with Ingress address
* 🎉

#### After
When user want to connect Global with Remote they have to
* Deploy Global
* Extract public IP of global and use it in Deploy Remote command
* 🎉

### Full changelog

* Introduced publicAddress and publicPort to Ingress data plane that indicates the address and port that is accessible to other clusters
* Removed flattening of Ingresses, therefore you don't have to have LB for them. Every ingress can have separate public IP
* On Kubernetes, automatically resolve publicAddress and publicPort. It works with both LoadBalancer services and NodePort.
  When LoadBalancer Service waits for its address, we still have an Ingress synced but without public coordinates, which means it won't be used.
  When LoadBalancer Service receives IP, it's automatically updated by Kubernetes controlled, synced to Global, and synced to other Remotes.
  You can override the address and port with annotations if you wish.
* Zone is now automatically created when Remote initiate a stream to Global (which is symmetric with DP that connects to CP creating Dataplane definition)
* Remove Ingress from zones tab in the GUI https://github.com/kumahq/kuma-gui/pull/81

### Documentation

- [ ] In progress

### CODE REVIEW

File: pkg/core/resources/apis/mesh/dataplane_validator.go [hunk 1]
```diff
@@ -121,8 +121,26 @@ func validateIngressNetworking(networking *mesh_proto.Dataplane_Networking) vali
 			}
 		}
 	}
-	for i, ingressInterface := range networking.GetIngress().GetAvailableServices() {
-		p := path.Field("ingress").Field("availableService").Index(i)
+	err.Add(validateIngress(path.Field("ingress"), networking.Ingress))
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/mesh/dataplane_validator.go:124]: Moved

File: pkg/core/resources/apis/mesh/dataplane_validator.go [hunk 2]
```diff
@@ -121,8 +121,26 @@ func validateIngressNetworking(networking *mesh_proto.Dataplane_Networking) vali
...
+	return err
+}
+
+func validateIngress(path validators.PathBuilder, ingress *mesh_proto.Dataplane_Networking_Ingress) validators.ValidationError {
+	if ingress == nil {
+		return validators.ValidationError{}
+	}
+	var err validators.ValidationError
+	if (ingress.GetPublicAddress() == "") != (ingress.GetPublicPort() == 0) {
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/mesh/dataplane_validator.go:133]: Ok, split the validation to two

File: pkg/plugins/runtime/k8s/controllers/pod_controller.go [hunk 3]
```diff
@@ -231,6 +231,7 @@ func (r *PodReconciler) createOrUpdateDataplane(
 	return nil
 }
 
+// HELLOOOO
```
Comment (lobkovilya) [inline pkg/plugins/runtime/k8s/controllers/pod_controller.go:234]: Do we really need this? :D

Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/controllers/pod_controller.go:234]: haha :D removed

**Review by lobkovilya — APPROVED** [review]

**Review by nickolaev — APPROVED** [review]

**Pull request merged**